File: solutions/lua/pig-latin/1/pig-latin.lua

local function translate(word)
    local rule1 = word:match('^[aeiouAEIOU]') or
        word:match('^xr') or word:match('^yt')
    if rule1 then
        return word .. 'ay'
    end
    local rule3 = word:match('^[^aeiouAEIOU]*qu')
    if rule3 then
        return word:sub(#rule3 + 1) .. rule3 .. 'ay'
    end
    local rule4 = word:match('^([^aeiouAEIOU]+)y')
    if rule4 then
        return word:sub(#rule4 + 1) .. rule4 .. 'ay'
    end
    local rule2 = word:match('^[^aeiouAEIOU]*')
    if rule2 then
        return word:sub(#rule2 + 1) .. rule2 .. 'ay'
    end
end

return function(phrase)
    local result = ''
    for word in phrase:gmatch('%S+') do
        result = result .. translate(word) .. ' '
    end
    return result:sub(1, -2)
end

File: solutions/php/pig-latin/1/PigLatin.php
<?php
declare(strict_types=1);

function translateWord(string $word): string
{
    if (preg_match('/^[aeiouAEIOU]|^xr|^yt/', $word)) {
        return "{$word}ay";
    }
    $rule3 = preg_match('/^[^aeiouAEIOU]*qu/', $word, $matches);
    if ($rule3) {
        $prefix = $matches[0];
        $rest = substr($word, strlen($prefix));
        return "{$rest}{$prefix}ay";
    }
    $rule4 = preg_match('/^([^aeiouAEIOU]+)y/', $word, $matches);
    if ($rule4) {
        $prefix = $matches[1];
        $rest = substr($word, strlen($prefix));
        return "{$rest}{$prefix}ay";
    }
    $rule2 = preg_match('/^[^aeiouAEIOU]+/', $word, $matches);
    if ($rule2) {
        $prefix = $matches[0];
        $rest = substr($word, strlen($prefix));
        return "{$rest}{$prefix}ay";
    }
    return "{$word}ay";
}

function translate(string $text): string
{
    $words = explode(' ', $text);
    return implode(' ', array_map('translateWord', $words));
}
